src/node/LaunchOptions.ts

```diff
@@ -13,30 +13,110 @@
  *
  * @public
  */
-export interface ChromeArgOptions {
+export interface BrowserArgOptions {
+  /**
+   * Whether to run the browser in headless mode.
+   * @defaultValue true
+   */
   headless?: boolean;
-  args?: string[];
+  /**
+   * Path to a user data directory.
+   * {@link https://chromium.googlesource.com/chromium/src/+/master/docs/user_data_dir.md | see the Chromium docs}
+   * for more info.
+   */
   userDataDir?: string;
+  /**
+   * Whether to auto-open a DevTools panel for each tab. If this is set to
+   * `true`, then `headless` will be set to `false` automatically.
+   * @defaultValue `false`
+   */
   devtools?: boolean;
+  /**
+   * Additional command line arguments to pass to the browser instance.
+   */
+  args?: string[];
 }
 
 /**
  * Generic launch options that can be passed when launching any browser.
  * @public
  */
 export interface LaunchOptions {
+  /**
+   * Path to a browser executable to use instead of the bundled Chromium. Note
+   * that Puppeteer is only guaranteed to work with the bundled Chromium, so use
+   * this setting at your own risk.
+   */
   executablePath?: string;
+  /**
+   * If `true`, do not use `puppeteer.defaultArgs()` when creating a browser. If
+   * an array is provided, these args will be filtered out. Use this with care -
+   * you probably want the default arguments Puppeteer uses.
+   * @defaultValue false
+   */
   ignoreDefaultArgs?: boolean | string[];
+  /**
+   * Close the browser process on `Ctrl+C`.
+   * @defaultValue `true`
+   */
   handleSIGINT?: boolean;
+  /**
+   * Close the browser process on `SIGTERM`.
+   * @defaultValue `true`
+   */
   handleSIGTERM?: boolean;
+  /**
+   * Close the browser process on `SIGHUP`.
+   * @defaultValue `true`
+   */
   handleSIGHUP?: boolean;
+  /**
+   * Maximum time in milliseconds to wait for the browser to start.
+   * Pass `0` to disable the timeout.
+   * @defaultValue 30000 (30 seconds).
+   */
   timeout?: number;
+  /**
+   * If true, pipes the browser process stdout and stderr to `process.stdout`
+   * and `process.stderr`.
+   * @defaultValue false
+   */
   dumpio?: boolean;
+  /**
+   * Specify environment variables that will be visible to the browser.
+   * @defaultValue The contents of `process.env`.
+   */
   env?: Record<string, string | undefined>;
+  /**
+   * Connect to a browser over a pipe instead of a WebSocket.
+   * @defaultValue false
+   */
   pipe?: boolean;
+  /**
+   * Which browser to launch.
+   * @defaultValue `chrome`
+   */
+  product?: Product;
+  /**
+   * {@link https://developer.mozilla.org/en-US/docs/Mozilla/Preferences/Preference_reference | Additional preferences } that can be passed when launching with Firefox.
+   */
+  extraPrefsFirefox?: Record<string, unknown>;
 }
+
+/**
+ * Utility type exposed to enable users to define options that can be passed to
+ * `puppeteer.launch` without having to list the set of all types.
+ * @public
+ */
+export type AllNodeLaunchOptions = BrowserArgOptions &
```

Comment:
I'm trying to find better names here because `BrowserArgOptions` and `BrowserOptions` are very similar.

I'm wondering about:

* `BrowserArgOptions` => `BrowserLaunchArgumentOptions` 
* `BrowserOptions` => `BrowserConnectOptions`

To make it clearer that `BrowserOptions` can be used when connecting to an existing browser (it's what `puppeteer.connect` takes) and `BrowserArgOptions` is only when launching browsers. 

@mathiasbynens wdyt? If we have to make a breaking change for this we might as well try to make all the names useful.

Reply:
`BrowserLaunchArgumentOptions` + `BrowserConnectOptions` sgtm

Reply:
Agree with Mathias that `BrowserLaunchArgumentOptions ` + `BrowserConnectOptions` looks reasonable.

`LaunchOptions` and `AllNodeLaunchOptions` have confusing names as well.

Reply:
Yeah these are confusing :( My hope is that users don't have to worry about the "sub types" and can just use the main type. I'd like to do a wider clean up of these types but they are used in a few places across the codebase and API so I'd rather keep the scope down for now and try to improve the names.

Reply:
SGTM